packages/playwright-core/src/server/common/selectorParser.ts

```diff
@@ -77,18 +77,20 @@ export function splitSelectorByFrame(selectorText: string): ParsedSelector[] {
     chunk.parts.push(part);
   }
   if (!chunk.parts.length)
-    throw new Error(`Selector cannot end with "content-frame", while parsing selector ${selectorText}`);
+    throw new Error(`Selector cannot end with entering frame, while parsing selector ${selectorText}`);
   result.push(chunk);
   if (typeof selector.capture === 'number' && typeof result[result.length - 1].capture !== 'number')
-    throw new Error(`Can not capture the selector before diving into the frame. Only use * after the last "content-frame"`);
+    throw new Error(`Can not capture the selector before diving into the frame. Only use * after the last frame has been selected`);
   return result;
 }
 
-
 export function stringifySelector(selector: string | ParsedSelector): string {
   if (typeof selector === 'string')
     return selector;
-  return selector.parts.map((p, i) => `${i === selector.capture ? '*' : ''}${p.name}=${p.source}`).join(' >> ');
+  return selector.parts.map((p, i) => {
+    const prefix = p.name === 'css' ? '' : p.name + '=';
+    return `${i === selector.capture ? '*' : ''}${prefix}${p.source}`;
```

Comment:
To be 100% correct, we have to escape quotes in the source, and also escape ">>" signs. If we never re-parse the result of this function, this might be ok.

Reply:
We can rename the method yes, we never re-parse.